lib/Service/ProviderConfigService.php

```diff
@@ -0,0 +1,100 @@
+<?php
+/**
+ * Nextcloud - ContextChat
+ *
+ * This file is licensed under the Affero General Public License version 3 or
+ * later. See the COPYING file.
+ *
+ * @author Anupam Kumar
+ * @copyright Anupam Kumar 2024
+ */
+
+namespace OCA\ContextChat\Service;
+
+use OCA\ContextChat\AppInfo\Application;
+use OCP\IConfig;
+
+/* array<[$appId__$providerId], $isInitiated> */
+
+class ProviderConfigService {
+	public function __construct(
+		private IConfig $config,
+	) {
+	}
+
+	public function getConfigKey(string $appId, string $providerId): string {
+		return $appId . '__' . $providerId;
```

Comment:
providerIds should technically be unique already

Reply:
yeah that is the unsaid rule but doing this would be more foolproof since the app makers would not need to specifically make it unique and we would not need to check for errors in case there are conflicts. The appId automatically makes it unique and error/worry free.